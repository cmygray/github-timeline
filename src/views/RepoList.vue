<template>
  <v-container tag="section" class="repo-list">
    <v-timeline>
      <v-timeline-item v-for="repo in repositories" :key="repo.id" small left>
        <v-card>
          <v-card-title>{{ repo.name }}</v-card-title>

          <v-card-text>
            <p>
              {{ repo.description }}
            </p>

            <v-chip class="mr-2">
              <v-icon small>star</v-icon>
              {{ repo.stargazers.totalCount }}
            </v-chip>

            <v-chip v-if="repo.isFork">
              <v-icon small>call_split</v-icon>
              Forked
            </v-chip>
          </v-card-text>
        </v-card>

        <template v-slot:opposite>{{ repo.createdAt | date }}</template>
      </v-timeline-item>
    </v-timeline>
  </v-container>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { Getter } from 'vuex-class';
import { Repository } from '@/models';

@Component
export default class Home extends Vue {
  @Prop()
  userId: string;

  @Getter('repositoriesByOwnerId')
  getRepositoriesByOwnerId: (userId: string) => Repository[];

  get repositories() {
    return this.getRepositoriesByOwnerId(this.userId);
  }
}
</script>
