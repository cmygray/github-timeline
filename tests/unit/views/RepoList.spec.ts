import { createLocalVue, mount } from '@vue/test-utils';
import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import RepoList from '@/views/RepoList.vue';
import { repositories } from '../__fixtures__/repositories';

Vue.use(Vuetify);

const localVue = createLocalVue();
localVue.use(Vuex);
// @ts-ignore-next-lineA
const vuetify = new Vuetify();

describe('Home.vue', () => {
  const USER_ID = 'cmygray';

  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(RepoList, {
      localVue,
      store: new Vuex.Store({
        getters: {
          repositoriesByOwnerId: () => () => repositories
        }
      }),
      vuetify,
      propsData: {
        userId: USER_ID
      }
    });
  });

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
