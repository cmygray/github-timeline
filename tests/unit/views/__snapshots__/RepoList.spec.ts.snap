// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders as expected 1`] = `
<section class="container repo-list">
  <div class="v-timeline theme--light">
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">type-graphql</div>
          <div class="v-card__text">
            <p>
              Create GraphQL schema and resolvers with TypeScript, using classes and decorators!
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-06-10T08:44:44Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">github-timeline</div>
          <div class="v-card__text">
            <p>

            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span>
            <!---->
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-06-09T07:19:12Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">vue-boilerplate</div>
          <div class="v-card__text">
            <p>
              A boilerplate for Vue.js using typescript, jest, router, Vuex...
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span>
            <!---->
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-06-07T07:53:37Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">vue-authenticate</div>
          <div class="v-card__text">
            <p>
              Simple Vue.js authentication library
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-04-25T11:06:37Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">jest</div>
          <div class="v-card__text">
            <p>
              Delightful JavaScript Testing.
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-04-18T12:42:29Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">reststate-client</div>
          <div class="v-card__text">
            <p>

            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-04-02T05:16:18Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">mobx-state-tree</div>
          <div class="v-card__text">
            <p>
              Model Driven State Management
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-02-11T09:05:16Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">autotrack</div>
          <div class="v-card__text">
            <p>
              Automatic and enhanced Google Analytics tracking for common user interactions on the web.
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2019-01-18T01:49:29Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">pyconkr-web</div>
          <div class="v-card__text">
            <p>
              üë©‚Äçüíª üë®‚Äçüíª This is a repository for developing pyconkr frontend.
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            18
          </span></span>
            <!---->
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2018-12-09T05:00:34Z</div>
    </div>
    <div class="v-timeline-item v-timeline-item--before theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">vscode-dash</div>
          <div class="v-card__text">
            <p>
              Dash and Zeal documentation integration in Visual Studio Code
            </p> <span class="mr-2 v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">star</i>
            0
          </span></span> <span class="v-chip v-chip--no-color theme--light v-size--default"><span class="v-chip__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">call_split</i>
            Forked
          </span></span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--small">
          <div class="v-timeline-item__inner-dot primary"></div>
        </div>
      </div>
      <div class="v-timeline-item__opposite">2018-10-07T07:46:13Z</div>
    </div>
  </div>
</section>
`;
